### PR TITLE
[receiver/splunkhec] Fix the double encoding of JSON responses

### LINKED ISSUE #27604: Double encoding of JSON responses
### Component(s)

receiver/splunkhec

## Steps to Reproduce
* Run the collector with splunkhecreceiver
* Send malformed data

## Expected Result
The result should be a JSON object that exposes the error message related to the handling of the data.

## Actual Result
The result is a JSON string representation of the JSON object.

### Collector version

0.87.0

### Environment information

_No response_

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
We would double-encode JSON objects as strings. We typically have encoded strings as JSON strings as static responses to issues encountered during data processing. Along the way, we adopted JSON objects to match the Splunk Enterprise API. However, we didn't stop encoding as JSON strings, resulting in double encoded JSON strings.

**Link to tracking Issue:**
Fixes #27604 

**Testing:**
Unit test changes.

### CODE REVIEW

**Review by fatsheep9146 — APPROVED** [review]

File: receiver/splunkhecreceiver/receiver.go [hunk 1]
```diff
@@ -58,11 +58,11 @@ var (
...
-	eventBlankRespBody        = initJSONResponse(responseErrEventBlank)
+	okRespBody                = []byte(responseOK)
+	eventRequiredRespBody     = []byte(responseErrEventRequired)
+	eventBlankRespBody        = []byte(responseErrEventBlank)
 	invalidEncodingRespBody   = initJSONResponse(responseInvalidEncoding)
-	invalidFormatRespBody     = initJSONResponse(responseInvalidDataFormat)
+	invalidFormatRespBody     = []byte(responseInvalidDataFormat)
 	invalidMethodRespBody     = initJSONResponse(responseInvalidMethod)
 	errGzipReaderRespBody     = initJSONResponse(responseErrGzipReader)
 	errUnmarshalBodyRespBody  = initJSONResponse(responseErrUnmarshalBody)
```
Comment (dmitryax) [inline receiver/splunkhecreceiver/receiver.go:68]: Can we update all the strings to the json format and remove `initJSONResponse` to be consistent?

Comment (atoulme) [reply receiver/splunkhecreceiver/receiver.go:68]: I guess. This has been ugly forever, true.

Comment (atoulme) [reply receiver/splunkhecreceiver/receiver.go]: Done!

**Comment by atoulme** [comment]
@dmitryax I made changes per your request, can we merge this?

**Review by dmitryax — APPROVED** [review]

**Pull request merged**